docs/linalg.rst

```diff
@@ -17,4 +17,145 @@ Matrices
 - ``R, S = ti.polar_decompose(A, ti.f32)``
 - ``U, sigma, V = ti.svd(A, ti.f32)`` (Note that ``sigma`` is a ``3x3`` diagonal matrix)
 
-TODO: doc here better like Vector.
+TODO: doc here better like Vector. WIP
+
+A matrix in Taichi can have two forms:
+
+  - as a temporary local variable. An ``n by m`` matrix consists of ``n * m`` scalar values.
+  - as a an element of a global tensor. In this case, the tensor is an ND array of ``n by m`` matrices.
```

Comment:
Would `N-dimensional` be easier to understand than `ND`?

Reply:
I like N-dimensional, lets go with that.